### PR TITLE
refactor(form-plugin): remove duplicate code

### PR DESCRIPTION
**Pull request by arturovt**
## PR Checklist
- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[ ] Feature
[x] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```
## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by arturovt** [comment]
@markwhitfeld followed your recommendations according to the last PR, just excluded nested `if`s/duplicate code and added distinctUntilChanged for valueChanges stream :blush:

**Comment by splincode** [comment]
@arturovt 
image

File: packages/form-plugin/src/directive.ts [hunk 1]
```diff
@@ -27,102 +38,95 @@ export class FormDirective implements OnInit, OnDestroy {
...
+        this.form.enable();
+      }
+
+      this._cd.markForCheck();
+    });
 
     this._formGroupDirective
       .valueChanges!.pipe(
         debounceTime(this.debounce),
+        distinctUntilChanged((a, b) => JSON.stringify(a) === JSON.stringify(b)),
```
Comment (markwhitfeld) [inline packages/form-plugin/src/directive.ts]: This looks like a bug fix as opposed to a refactor. Could this be in a separate PR?

File: packages/form-plugin/src/directive.ts [hunk 2]
```diff
@@ -27,102 +38,95 @@ export class FormDirective implements OnInit, OnDestroy {
...
+                path: this.path,
+                dirty: this._formGroupDirective.dirty
+              }),
+              new UpdateFormErrors({
+                path: this.path,
+                errors: this._formGroupDirective.errors
+              })
+            ])
+            .pipe(finalize(() => (this._updating = false)));
+        }),
```
Comment (markwhitfeld) [inline packages/form-plugin/src/directive.ts]: I think that the previous `subscribe` version (as opposed to the `mergeMap` and `finalize`) is clearer, simpler code with less rxjs dependencies.

File: packages/form-plugin/src/directive.ts [hunk 3]
```diff
@@ -27,102 +38,95 @@ export class FormDirective implements OnInit, OnDestroy {
...
-            error: () => (this._updating = false),
-            complete: () => (this._updating = false)
-          });
-      });
+      .subscribe();
 
     this._formGroupDirective
       .statusChanges!.pipe(
         debounceTime(this.debounce),
+        distinctUntilChanged(),
```
Comment (markwhitfeld) [inline packages/form-plugin/src/directive.ts]: Could also be in the bugfix PR I mentioned for the other distinctUntilChanged operator.

**Comment by arturovt** [comment]
@markwhitfeld ping :rocket:

File: packages/form-plugin/src/directive.ts (hunk 2 above)
Comment (splincode) [reply packages/form-plugin/src/directive.ts]: What is the difference, how many dependencies do you have? If they are still under the Angular hood and somehow get into the bundle...

Savings on matches, against readable code

**Comment by markwhitfeld** [comment]
@arturovt could you investigate why the coverage dropped?

**Comment by arturovt** [comment]
@markwhitfeld just installed codeclimate's browser extension that shows more detailed info about diff-coverage - and it showed nothing :smile: maybe you could know what is wrong - i'd fix it?

**Comment by arturovt** [comment]
@markwhitfeld ping :rocket: 
just added some tests that didn't cover runtime behavior

**Comment by markwhitfeld** [comment]
Approved the changes in code climate
